packages/runtime/lib/worker/message-port-writable.js

```diff
@@ -12,28 +12,24 @@ class MessagePortWritable extends Writable {
   }
 
   _write (chunk, encoding, callback) {
-    this.port.postMessage({
-      metadata: this.metadata,
-      logs: [chunk.toString().trim()],
-    })
+    this.port.postMessage({ metadata: this.metadata, logs: [chunk.toString().trim()] })
     process.nextTick(callback)
   }
 
   _writev (chunks, callback) {
     // Process the logs here before trying to send them across the thread
     // boundary. Sometimes the chunks have an undocumented method on them
     // which will prevent sending the chunk itself across threads.
-    const logs = chunks.map((chunk) => {
-      // pino should always produce a string here.
-      assert(typeof chunk.chunk === 'string')
-      return chunk.chunk.trim()
-    })
-
-    this.port.postMessage({
-      metadata: this.metadata,
-      logs,
-    })
-    setImmediate(callback)
+    const logs = []
+
+    for (const { chunk } of chunks) {
+      if (typeof chunk === 'string') {
+        logs.push(chunk.trim())
+      }
+    }
+
+    this.port.postMessage({ metadata: this.metadata, logs })
+    process.nextTick(callback)
```

Comment:
Use setImmediate

Reply:
Actually I realized we don't need nextTick in this file at all. Removed.